src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx

```diff
@@ -0,0 +1,283 @@
+    this.loadFieldValues(newValue);
+    this.props.onChange(newValue);
+  };
+
+  handleClickOutside = (event: MouseEvent) => {
+    if (
+      event.target instanceof HTMLElement &&
+      this.selectorField.current &&
+      this.selectorField.current.contains(event.target)
+    ) {
+      return;
+    }
+
+    this.setState({
+      showOptions: false,
+    });
+  };
+
+  handleClickSuggestionItem = (suggestion: Suggestion) => () => {
+    let fieldValues = [...this.state.fieldValues];
+    const lastFieldValue = fieldValues[fieldValues.length - 1];
+
+    if (defined(lastFieldValue)) {
+      fieldValues[fieldValues.length - 1] = suggestion;
+    }
+
+    if (!defined(lastFieldValue)) {
+      fieldValues = [suggestion];
+    }
+
+    this.setState(
+      {
+        fieldValues,
+        showOptions: false,
+      },
+      () => {
+        this.handleChangeParentValue();
+      }
+    );
+  };
+
+  handleChangeParentValue = () => {
+    const {onChange} = this.props;
+    const {fieldValues} = this.state;
+    const newValue: Array<string> = [];
+
+    for (const index in fieldValues) {
+      const fieldValue = fieldValues[index];
+      if (typeof fieldValue !== 'string') {
+        newValue.push(fieldValue.value);
+        continue;
+      }
+      newValue.push(fieldValue);
+    }
+
+    onChange(newValue.join(' '));
+  };
+
+  handleFocus = () => {
+    this.setState({
+      showOptions: true,
+    });
+  };
+
+  handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const {fieldValues} = this.state;
+
+    if (event.keyCode === 32) {
+      this.setState({
+        fieldValues: [...fieldValues, ' '],
+      });
+    }
+  };
+
+  render() {
+    const {error, onBlur, disabled, value} = this.props;
+    const {showOptions, suggestions} = this.state;
+
+    return (
+      <Wrapper ref={this.selectorField}>
+        <StyledTextField
+          name="from"
+          placeholder={t('ex. strings, numbers, custom')}
+          onChange={this.handleChange}
+          autoComplete="off"
+          autoFocus
+          value={value}
+          onKeyDown={this.handleKeyDown}
+          error={error}
+          onBlur={onBlur}
+          onFocus={this.handleFocus}
+          disabled={disabled}
+        />
+        {showOptions && suggestions.length > 0 && (
+          <SuggestionsWrapper>
+            {suggestions.map(suggestion => (
+              <SuggestionItemWrapper
+                key={suggestion.value}
+                onClick={this.handleClickSuggestionItem(suggestion)}
+              >
+                <Tooltip
+                  title={`${suggestion.value} ${suggestion?.description &&
+                    `(${suggestion.description})`}`}
+                  position="top"
+                >
+                  <SuggestionItem>
+                    <TextOverflow>{suggestion.value}</TextOverflow>
+                    {suggestion?.description && (
+                      <SuggestionDescription>
```

Comment:
@doralchan was suggesting to make the description gray.

Reply:
Good idea 💡